src/amo/reducers/recommendations.js

```diff
@@ -0,0 +1,182 @@
+/* @flow */
+import invariant from 'invariant';
+
+import { createInternalAddon } from 'core/reducers/addons';
+import type { ExternalAddonType } from 'core/types/addons';
+
+export const ABORT_FETCH_RECOMMENDATIONS: 'ABORT_FETCH_RECOMMENDATIONS'
+  = 'ABORT_FETCH_RECOMMENDATIONS';
+export const FETCH_RECOMMENDATIONS: 'FETCH_RECOMMENDATIONS'
+  = 'FETCH_RECOMMENDATIONS';
+export const LOAD_RECOMMENDATIONS: 'LOAD_RECOMMENDATIONS'
+  = 'LOAD_RECOMMENDATIONS';
+
+export type Recommendations = {|
+  addons: Array<ExternalAddonType> | null,
+  fallbackReason: string | null,
+  loading: boolean,
+  outcome: string | null,
+|};
+
+export type RecommendationsState = {|
+  byGuid: {
+    [guid: string]: Recommendations,
+  },
+|};
+
+export const initialState: RecommendationsState = {
+  byGuid: {},
+};
+
+export type AbortFetchRecommendationsParams = {|
+  guid: string,
+|};
+
+type AbortFetchRecommendationsAction = {|
+  type: typeof ABORT_FETCH_RECOMMENDATIONS,
+  payload: AbortFetchRecommendationsParams,
+|};
+
+export const abortFetchRecommendations = ({
+  guid,
+}: AbortFetchRecommendationsParams = {}): AbortFetchRecommendationsAction => {
+  invariant(guid, 'guid is required');
+  return {
+    type: ABORT_FETCH_RECOMMENDATIONS,
+    payload: { guid },
+  };
+};
+
+type FetchRecommendationsParams = {|
+  errorHandlerId: string,
+  guid: string,
+  recommended: boolean,
+|};
+
+export type FetchRecommendationsAction = {|
+  type: typeof FETCH_RECOMMENDATIONS,
+  payload: FetchRecommendationsParams,
+|};
+
+export const fetchRecommendations = ({
+  errorHandlerId,
+  guid,
+  recommended,
+}: FetchRecommendationsParams = {}): FetchRecommendationsAction => {
+  invariant(errorHandlerId, 'errorHandlerId is required');
+  invariant(guid, 'guid is required');
+  invariant(typeof recommended === 'boolean', 'recommended is required');
+
+  return {
+    type: FETCH_RECOMMENDATIONS,
+    payload: { errorHandlerId, guid, recommended },
+  };
+};
+
+export type LoadRecommendationsParams = {|
+  addons: Array<ExternalAddonType>,
+  fallbackReason: string,
+  guid: string,
+  outcome: string,
+|};
+
+type LoadRecommendationsAction = {|
+  type: typeof LOAD_RECOMMENDATIONS,
+  payload: LoadRecommendationsParams,
+|};
+
+export const loadRecommendations = ({
+  addons,
+  fallbackReason,
+  guid,
+  outcome,
+}: LoadRecommendationsParams = {}): LoadRecommendationsAction => {
+  invariant(addons, 'addons is required');
+  invariant(fallbackReason, 'fallbackReason is required');
```

Comment:
The API documentation says the `fallback_reason` is "nullable": http://addons-server.readthedocs.io/en/latest/topics/api/addons.html#recommendations

Reply:
Ah, correct, good call!